src/lib/rc/dsm.cpp

```diff
@@ -470,6 +470,7 @@ dsm_decode(hrt_abstime frame_time, uint16_t *values, uint16_t *num_values, bool
 		 * Specifically, the first four channels in rc_channel_data are roll, pitch, thrust, yaw,
 		 * but the first four channels from the DSM receiver are thrust, roll, pitch, yaw.
 		 */
+#if !defined(__PX4_QURT)
```

Comment:
I believe at calibration time it assigns each stick axis to a channel. So I'm thinking if you left this alone and recalibrated it should work fine and be consistent with the rest of the system. Why the channels are swapped in the first place is another question, but at least it's consistent.